src/zarr/core/metadata/v3.py

```diff
@@ -606,6 +607,10 @@ def from_numpy(cls, dtype: np.dtype[Any]) -> DataType:
             return DataType.string
         elif dtype.kind == "S":
             return DataType.bytes
+        elif not _NUMPY_SUPPORTS_VLEN_STRING and dtype.kind == "O":
+            # numpy < 2.0 does not support vlen string dtype
+            # so we fall back on object array of strings
+            return DataType.string
```

Comment:
Do we support objects other than strings in v3?

Reply:
No, only `string` and `bytes`. But they seem to have another `kind`